BEASTIE/extractHets.py

```diff
@@ -25,55 +39,98 @@ def count_all_het_sites(tmp,sample,vcfFilename,file_dir,outputFilename,chr_start
-                            pos=fields[1]                         # column 9
-                            rs = fields[2]                        # column 10
-                            genotype = fields[9].split(':')[0]
-                            if(not isHeterozygous(str(genotype))):continue # go back to the begining of the loop
-                            transcriptCoord=transcript.mapToTranscript(int(pos))#
-                            total_biSNP += 1
-                            chr_pos=chromN+"_"+pos
-                            byGene[geneID].add(chr_pos)
-                            out_stream.write("\t".join([str(chrom),str(chromN),str(geneID),str(pos),str(transID),str(transcriptCoord),str(rs),str(genotype),"\n"]))
-            # write up the basic information
-            # all_set_list = []
-            # non_empty_count = 0
-            # non_empty_count_len = 0
-            # for each_key in byGene:
-            #     all_set_list.extend(list(byGene[each_key]))
-            #     if len(byGene[each_key])!= 0:
-            #         non_empty_count += 1
-            #         non_empty_count_len += len(byGene[each_key])
+
+                        region_str = f"{chromN}:{begin}-{end}"
+                        if not region_str in region_str_to_transcripts:
+                            region_str_to_transcripts[region_str] = []
+                        # else:
+                        #     for existing in region_str_to_transcripts[region_str]:
+                        #         print(f"existing transcript @ {region_str} {existing.getTranscriptId()} != {transcript.getTranscriptId()}")
+                        region_str_to_transcripts[region_str].append(transcript)
+            
+            print(f"{datetime.now()} found {len(region_str_to_transcripts)} regions to process")
+            CHUNK_SIZE = 1000
+            outputs = []
+            for x in chunk_iter(iter(region_str_to_transcripts.keys()), CHUNK_SIZE):
+                regions = ' '.join(x)
+                cmd = f"tabix --separate-regions {vcfFilename} {regions}"
+                output = Pipe.run(cmd)
+                if len(output) > 0:
+                    outputs.append(output)
+
+            print(f"{datetime.now()} got {len(outputs)} tabix call outputs")
+
+            out_stream = open(outputFile, "w")
+            out_stream.write("chr\tchrN\tgeneID\tpos\ttranscriptID\ttranscript_pos\tSNP_id\tgenotype\n")
+            for output in outputs:
+                lines = output.split("\n")
+                transcripts = None
+                for line in lines:
+                    if line.startswith('#'):
+                        region_str = line[1:]
+                        if region_str not in region_str_to_transcripts:
+                            print(f"region_str not found! {region_str}")
+                        transcripts = region_str_to_transcripts[region_str]
+                        continue
+
+                    assert transcript is not None
```

Comment:
What is this for? I feel this assert depends on an interaction between lines 55 and line 113. is it supposed to be 
```suggestion
                    assert transcripts is not None
```